src/osd/scrubber/scrub_reservations.h

```diff
@@ -140,6 +168,17 @@ class ReplicaReservations {
    */
   bool send_next_reservation_or_complete();
 
+  /// (internal helper) is this is a reply to our last request?
+  tl::expected<bool, std::string> is_response_relevant(
+      epoch_t msg_epoch,
+      reservation_nonce_t msg_nonce,
+      pg_shard_t from) const;
+
+  /// (internal helper) is this reply coming from the expected replica?
+  tl::expected<bool, std::string> is_msg_source_correct(
```

Comment:
The intended design should make this impossible, right?  That needs to be explained here.